### ISSUE TITLE
[BUG]:Tests are executing locally instead of in cloud browsers.

### ISSUE THREAD

**Issue by raju-a** [microsoft/playwright-testing-service#82]
**Describe the bug**
Early on, we were running our playwright service test with an old configuration that included playwright-service.js, Now we have created a new subscription and workspace.
As per the document, we have .env file and configuration

*.env*
```
PLAYWRIGHT_SERVICE_ACCESS_KEY=OUR_TOKEN
PLAYWRIGHT_SERVICE_URL=OUR_URL
```

*playwright.service.config.js*

```javascript
 /*
* This file enables Playwright client to connect to remote browsers.
* It should be placed in the same directory as playwright.config.ts.
*/

import { defineConfig } from '@playwright/experimental-ct-react';
import config from './playwright-ct.config';
import dotenv from 'dotenv';

// Define environment on the dev box in .env file:
//  .env:
//    PLAYWRIGHT_SERVICE_ACCESS_TOKEN=XXX
//    PLAYWRIGHT_SERVICE_URL=XXX

// Define environment in your GitHub workflow spec.
//  env:
//    PLAYWRIGHT_SERVICE_ACCESS_TOKEN: ${{ secrets.PLAYWRIGHT_SERVICE_ACCESS_TOKEN }}
//    PLAYWRIGHT_SERVICE_URL: ${{ secrets.PLAYWRIGHT_SERVICE_URL }}
//    PLAYWRIGHT_SERVICE_RUN_ID: ${{ github.run_id }}-${{ github.run_attempt }}-${{ github.sha }}

dotenv.config();

// Name the test run if it's not named yet.
process.env.PLAYWRIGHT_SERVICE_RUN_ID = process.env.PLAYWRIGHT_SERVICE_RUN_ID || new Date().toISOString();

// Can be 'linux' or 'windows'.
const os = process.env.PLAYWRIGHT_SERVICE_OS || 'linux';

export default defineConfig(config, {
  // Define more generous timeout for the service operation if necessary.
  // timeout: 60000,
  // expect: {
  //   timeout: 10000,
  // },
  workers: 20,

  // Enable screenshot testing and configure directory with expectations.
  // https://learn.microsoft.com/azure/playwright-testing/how-to-configure-visual-comparisons
  ignoreSnapshots: false,
  snapshotPathTemplate: `{testDir}/__screenshots__/{testFilePath}/${os}/{arg}{ext}`,
  
  use: {
    // Specify the service endpoint.
    connectOptions: {
      wsEndpoint: `${process.env.PLAYWRIGHT_SERVICE_URL}?cap=${JSON.stringify({
        // Can be 'linux' or 'windows'.
        os,
        runId: process.env.PLAYWRIGHT_SERVICE_RUN_ID
      })}`,
      timeout: 30000,
      headers: {
        'x-mpt-access-key': process.env.PLAYWRIGHT_SERVICE_ACCESS_TOKEN
      },
      // Allow service to access the localhost.
      exposeNetwork: 'localhost'
    }
  },
  // Tenmp workaround for config merge bug in OSS https://github.com/microsoft/playwright/pull/28224
  projects: config.projects? config.projects : [{}]
});

```
*playwright-ct.config.js*

```javascript
// @ts-check
const { defineConfig, devices } = require('@playwright/experimental-ct-react');
const path = require('path');

/**
 * @see https://playwright.dev/docs/test-configuration
 */
module.exports = defineConfig({
    testDir: './test/UI/functional-component/',
    /* The base directory, relative to the config file, for snapshot files created with toMatchSnapshot and toHaveScreenshot. */
    snapshotDir: './__snapshots__',
    /* Maximum time one test can run for. */
    /* Increased the timeout to 30 seconds due to playwright-service page setup issue. 
    https://github.com/microsoft/playwright-testing-service/issues/59 */
    timeout: 30 * 1000,
    /* Run tests in files in parallel */
    fullyParallel: true,
    /* Fail the build on CI if you accidentally left test.only in the source code. */
    forbidOnly: !!process.env.CI,
    /* Retry on CI only */
    retries: process.env.CI ? 2 : 0,
    /* Opt out of parallel tests on CI. */
    workers: process.env.CI ? 1 : undefined,
    /* Reporter to use. See https://playwright.dev/docs/test-reporters */
    reporter: 'html',
    /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
    use: {
        /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
        trace: 'retain-on-failure',
        headless: false,
        /* Port to use for Playwright component endpoint. */
        ctPort: 3100,
        ctViteConfig: {
            resolve: {
                alias: [
                    { find: './web-worker', replacement: './web-worker?worker' },
               
                ],
            },
            define: {
                logger: 'window.console',
            },
        },
    },

    /* Configure projects for major browsers */
    projects: [
        {
            name: 'chromium',
            use: { ...devices['Desktop Chrome'] },
        },
        {
            name: 'firefox',
            use: { ...devices['Desktop Firefox'] },
        },
        {
            name: 'webkit',
            use: { ...devices['Desktop Safari'] },
        },
    ],
});

```
**To Reproduce**
 ```playwright test -c playwright.service.config.js --workers=50```

**Expected behavior**
it should run tests in service

**Setup information (please complete the following information):**
 - Azure Subscription Id: 0ce4637a-94fc-461d-b601-f6fc4b559643
 - Workspace Name:
 - Service trace Id (if available):
 - Log snippet (if available):
 - Browser on which tests are executed (if applicable):
 - Time range (in PST) when the issue was experienced:
 - Playwright Version:  playwright/experimental-ct-react      1.40.1 
 - Client machine OS: mac os monterey
 - OS requested on Service: linux

AB#1923138

**Comment by puagarwa** [comment]
HI @raju-a, Did you tried this suggestion and able to run it ?

**Comment by puagarwa** [comment]
HI @Avinash-kumar99 That is strange and need to look into logs. 
- Can i assume that you are able to repro it from your laptop as well ? to just keep CI/CD integration aside and focus on service integration with test suite ?
- What i understood is, you are not getting any error but you feel tests are not executing against service? 
- To rule out env variable issue, can i ask you to set the env variable in config file itself, like this process.env.PLAYWRIGHT_SERVICE_ACCESS_TOKEN=token and process.env.PLAYWRIGHT_SERVICE_URL=url
- We can setup a call for quick resolution if above doesnt help.

**Comment by Avinash-kumar99** [comment]
Hi @puagarwa , 
- Yes, From local we are triggering it. In old subscription, once we trigger it, tests will execute in service and we would see the result in terminal but in new subscription, when we trigger, it is executing in local by opening the browsers and we are not getting any errors
[https://share.anysnap.app/fatgscEkE2GR](https://share.anysnap.app/fatgscEkE2GR)
- We had set the env variable in the config file, but we are still facing the same issue.
- Sure, we can schedule a call and we are available anytime

**Comment by puagarwa** [comment]
Hi @Avinash-kumar99, @raju-a I looked into two isues we observed in our session and below are updates

- Playwirght.service.config.js not able to run test in service mode: 
there is issue in type export in playwright-ct-react [package ](https://github.com/microsoft/playwright/blob/main/packages/playwright-ct-react/index.d.ts#L62) and hence it is not able to update config with service config option and test keep running in local mode. I will get it fixed in OSS but i am looking for workaround which i will share in 1-2 days.

- exposeNetwork not working in 1.40 but works in 1.39:
This issue actually happens for me in old version like 1.35 but with 1.39 and 1.40 it works fine, can you once again change version to 1.40 and try again, just wondering if it was due to package not really on 1.40 initially when we observe the issue.

**Comment by puagarwa** [comment]
Hi @Avinash-kumar99 , @raju-a , Below is workaround for "Playwirght.service.config.js not able to run test in service mode"
IN playwright.service.config.js you could import defineconfig from @playwright/test module instead of ct-react like this
import { defineConfig } from '@playwright/test';

**Comment by vvs11** [comment]
@raju-a does this solution work for you ?

**Comment by puagarwa** [comment]
@raju-a closing this issue as confirmed in call. feel free to open new issue if you face any issue.
